documents/invoice/service.go

```diff
@@ -107,7 +107,7 @@ func (s service) validateAndPersist(ctx context.Context, old, new documents.Mode
 	}
 
 	// we use CurrentVersion as the id since that will be unique across multiple versions of the same document
-	err = s.repo.Create(self.ID[:], inv.CoreDocument.CurrentVersion, inv)
+	err = s.repo.Create(self.ID[:], inv.CoreDocumentModel.Document.CurrentVersion, inv)
```

Comment:
See my above comment, where you call pack to get the DocumentIdentifier. This doesn't seem necessary?